srcJava/com/rti/perftest/harness/PerftestPrinter.java

```diff
@@ -0,0 +1,486 @@
+                        System.out.printf("Length (Bytes), Total Samples," +
+                                      " Ave Samples/s,    Ave Mbps, " +
+                                      "Lost Samples, Lost Samples (%%)");
+                        if (_showCPU) {
+                            System.out.printf(", CPU (%%)");
+                        }
+                        System.out.printf("\n");
+                    }
+                    System.out.printf("%1$14d,%2$14d,%3$14.0f,%4$12.1f," +
+                                  "%5$13d,%6$17.2f",
+                                  length,
+                                  intervalPacketsReceived,
+                                  intervalPacketsReceived * 1000000.0
+                                         / intervalTime,
+                                  intervalBytesReceived * 1000000.0
+                                         / intervalTime * 8.0 / 1000.0 / 1000.0,
+                                  intervalMissingPackets,
+                                  missingPacketsPercent);
+                    if (_showCPU) {
+                        System.out.printf(",%1$8.2f", outputCpu);
+                    }
+                    System.out.printf("\n");
+                    System.out.flush();
+                    break;
+                case JSON:
+                    if (_printIntervals) {
+                        System.out.printf("\n\t\t\t],\n");
+                    }
+                    System.out.printf("\t\t\t\"summary\":{\n" +
+                                  "\t\t\t\t\"packets\": %1$1d,\n" +
+                                  "\t\t\t\t\"packets/sAve\": %2$1.0f,\n" +
+                                  "\t\t\t\t\"mbpsAve\": %3$1.1f,\n" +
+                                  "\t\t\t\t\"lost\": %4$1d,\n" +
+                                  "\t\t\t\t\"lostPercent\": %5$1.2f",
+                                  intervalPacketsReceived,
+                                  intervalPacketsReceived * 1000000.0
+                                         / intervalTime,
+                                  intervalBytesReceived * 1000000.0
+                                         / intervalTime * 8.0 / 1000.0 / 1000.0,
+                                  intervalMissingPackets,
+                                  missingPacketsPercent);
+                    if (_showCPU) {
+                        System.out.printf(",\n\t\t\t\t\"cpu\": %1$1.2f", outputCpu);
+                    }
+                    System.out.printf("\n\t\t\t}\n\t\t}");
+                    break;
+                case LEGACY:
+                System.out.printf("Length: %1$5d  Packets: %2$8d  " +
+                                  "Packets/s(ave): %3$7.0f  " +
+                                  "Mbps(ave): %4$7.1f  Lost: %5$5d (%6$1.2f%%)",
+                                  length,
+                                  intervalPacketsReceived,
+                                  intervalPacketsReceived * 1000000.0 /
```

Comment:
I was thinking: Is there a reason why we don't do these calculations only once, keep them as local variables and then using them instead of doing the calculations for each case? This would apply to all the languages.

Reply:
We have to do these each time as function is called, so i don't understand you why do you want to save as local variables.